func/db.sh

```diff
@@ -129,6 +133,125 @@ mysql_dump() {
+	# Query the password policy variables.
+	local length=$(mysql --defaults-extra-file="$mycnf" -N -e "SHOW VARIABLES LIKE 'validate_password.length';" | awk '{print $2}')
+	local mixed_case_count=$(mysql --defaults-extra-file="$mycnf" -N -e "SHOW VARIABLES LIKE 'validate_password.mixed_case_count';" | awk '{print $2}')
+	local number_count=$(mysql --defaults-extra-file="$mycnf" -N -e "SHOW VARIABLES LIKE 'validate_password.number_count';" | awk '{print $2}')
+	local special_char_count=$(mysql --defaults-extra-file="$mycnf" -N -e "SHOW VARIABLES LIKE 'validate_password.special_char_count';" | awk '{print $2}')
+
+	# If any query fails, default values ​​are assigned
+	[ -z "$length" ] && length=8
+	[ -z "$mixed_case_count" ] && mixed_case_count=1
+	[ -z "$number_count" ] && number_count=1
+	[ -z "$special_char_count" ] && special_char_count=1
+
+	# Define character sets
+	local lower='abcdefghijklmnopqrstuvwxyz'
+	local upper='ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+	local digits='0123456789'
+	local specials='!@#$%^&*()-_=+[]{};:,.<>?/'
+
+	local p_lower=""
+	local p_upper=""
+	local p_digit=""
+	local p_special=""
+
+	# Generate at least mixed_case_count lowercase characters and mixed_case_count uppercase characters
+	for i in $(seq 1 $mixed_case_count); do
+		p_lower+=$(tr -dc "$lower" </dev/urandom | head -c 1)
+		p_upper+=$(tr -dc "$upper" </dev/urandom | head -c 1)
+	done
+
+	# Generate at least number_count digits
+	for i in $(seq 1 $number_count); do
+		p_digit+=$(tr -dc "$digits" </dev/urandom | head -c 1)
+	done
+
+	# Generate at least special_char_count special characters
+	for i in $(seq 1 $special_char_count); do
+		p_special+=$(tr -dc "$specials" </dev/urandom | head -c 1)
+	done
+
+	# Calculate the number of remaining characters
+	local mandatory_count=$((mixed_case_count * 2 + number_count + special_char_count))
+	local rest_length=$((length - mandatory_count))
+	if [ $rest_length -lt 0 ]; then
+		rest_length=0
+	fi
+
+	# Generate the remaining characters of any type
+	local all="$lower$upper$digits$specials"
+	local p_rest=$(tr -dc "$all" </dev/urandom | head -c $rest_length)
+
+	# Concatenate all parts
+	local password="$p_lower$p_upper$p_digit$p_special$p_rest"
+
+	# Shuffle randomly to avoid fixed patterns
+	echo "$password" | fold -w1 | shuf | tr -d '\n'
+	echo ""
+}
+
+create_mysql_user_with_policy() {
+	local err_file="/tmp/create_user.err"
+	rm -f "$err_file"
+
+	# Check if user already exists (regardless of host)
+	local user_exists
+	user_exists=$(mysql --defaults-extra-file="$mycnf" -N -e "SELECT COUNT(*) FROM mysql.user WHERE user='$DBUSER';")
+	if [ "$user_exists" -gt 0 ]; then
+		echo "User $DBUSER already exists. Skipping creation."
+		return 0
+	fi
+
+	local create_cmd
+
+	# Try to create the user using the backup password (MD5 hash)
+	# NOTE: We used "IDENTIFIED WITH mysql_native_password BY ..." to create the user
+	create_cmd="CREATE USER IF NOT EXISTS '$DBUSER'@'%' IDENTIFIED WITH mysql_native_password BY '$dbpass';"
```

Comment:
Where does it takes $dbpass get from "MD5" is hased / encrypted so can't be used. So in this we case we need to generate a new password first and then overwrite password.